lib/elixir/lib/gen_event.ex

```diff
@@ -380,8 +380,10 @@ defmodule GenEvent do
         :gen.start(GenEvent, mode, @no_callback, [], [])
       atom when is_atom(atom) ->
         :gen.start(GenEvent, mode, {:local, atom}, @no_callback, [], [])
-      other when is_tuple(other) ->
-        :gen.start(GenEvent, mode, other, @no_callback, [], [])
+      tuple when is_tuple(tuple) ->
+        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
+      other ->
+        raise StartLinkNameError, arg: other
```

Comment:
I don't we want to introduce a new exception for something that isn't going to be explicitly rescued. Perhaps we could raise an `ArgumentError` instead?